morpheus/_lib/include/morpheus/objects/rmm_tensor.hpp

```diff
@@ -144,6 +144,13 @@ class RMMTensor : public ITensor
      */
     std::shared_ptr<ITensor> as_type(DType dtype) const override;
 
+    /**
+     * @brief Get the underlying rmm buffer as-id and does not apply the offset.
+     *
+     * @return std::shared_ptr<rmm::device_buffer>
+     */
+    std::shared_ptr<rmm::device_buffer> get_buffer();
+
```

Comment:
This makes me nervous returning a `shared_ptr` to the original rmm buffer here. I feel like there has to be a better, and less error prone way. Lets disucss.

Reply:
Reverted